app/models/image_data.rb

```diff
@@ -47,4 +50,15 @@ def dimensions
                       Dimensions.new(image[:width], image[:height])
                     end
   end
+
+  def filename_is_unique
+    return if validate_on_edition.blank? || validate_on_image.blank? || file.blank?
+
+    edition = validate_on_edition
+    image = validate_on_image
```

Comment:
You shouldn't actually need the Edition object because the Image will already be associated with it.

```suggestion
    return if validate_on_image.blank? || file.blank?

    edition = validate_on_image.edition
    image = validate_on_image
```